src/Encoder.h

```diff
@@ -10,33 +17,59 @@ namespace cppkin
 {
     class Span;
     class SimpleAnnotation;
-    class EncoderContext;
 
-    template<EncodingTypes::Enumeration >
-    class Encoder
+    class Encoder {
```

Comment:
Encoder Interface should be private. no one needs to address the encoder beside EncoderContext in my opinion.
EncoderContext should be friend instead.

Reply:
What is your opinion on the need for the interface to be private and using friend in order to allow the Context to use the encoder?